library/Reporting/Web/Forms/TemplateForm.php

```diff
@@ -16,30 +16,47 @@ class TemplateForm extends CompatForm
 {
     use Database;
 
-    /** @var bool Hack to disable the {@link onSuccess()} code upon deletion of the template */
-    protected $callOnSuccess;
-
     protected $template;
 
     public function getTemplate()
     {
         return $this->template;
     }
 
-    public function setTemplate($template)
+    /**
+     * Creates a new `TemplateForm` instance with a given template
+     *
+     * @param $template
+     *
+     * @return TemplateForm
+     */
+    public static function fromTemplate($template): TemplateForm
     {
-        $this->template = $template;
+        $form = new static();
+
+        $form->template = $template;
 
         if ($template->settings) {
-            $this->populate(array_filter($template->settings, function ($value) {
+            $form->populate(array_filter($template->settings, function ($value) {
                 // Don't populate files
                 return ! is_array($value);
             }));
         }
 
-        return $this;
+        return $form;
+    }
+
+    /**
+     * @inheritDoc
+     */
```

Comment:
We use `inheritDoc` only when adding custom ones in addition to the parent docs.